common/amundsen_common/models/feature.py

```diff
@@ -0,0 +1,97 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import List, Optional, Dict, Any
+
+import attr
+
+from amundsen_common.models.user import User
+from amundsen_common.models.badge import Badge
+from amundsen_common.models.tag import Tag
+from amundsen_common.models.table import Column, ProgrammaticDescription, Watermark
+from marshmallow3_annotations.ext.attrs import AttrsSchema
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class Query:
```

Comment:
more generic, I tried to follow what we have for dashboards already because we are reusing the same component to show this code snippet or query in frontend

Reply:
we could rename this something more generic like code snippet or generation code so it goes with the common schema

Reply:
I just tagged you in the RFC thread here: https://github.com/amundsen-io/rfcs/pull/37/files#r632790084

Do you mind if we have the conversation on that thread and provide a synopsis here once resolved?

Reply:
For reference, this was discussed and resolved in another PR: https://github.com/amundsen-io/rfcs/pull/37#discussion_r633545054

Query was then moved to separate file.

Reply:
TLDR: made query file with generic and simple query entity that can be expanded on with query and query composition work